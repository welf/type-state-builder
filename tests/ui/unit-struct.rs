use type_state_builder::TypeStateBuilder;

// This should be an error: unit structs are not supported
#[derive(TypeStateBuilder)]
struct UnitStruct;

fn main() {}
