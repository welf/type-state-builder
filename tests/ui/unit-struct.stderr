error: TypeStateBuilder does not support unit structs
 --> tests/ui/unit-struct.rs:5:1
  |
5 | struct UnitStruct;
  | ^^^^^^^^^^^^^^^^^^
